app.ts

```diff
@@ -0,0 +1,187 @@
+//#region Global Variables
+let resizeTimer = 50;
+let previousCursor: number[];
+let previousScale: number;
+
+//#endregion Global Variables
+
+//#region AppLogic
+//#region API
+async function getRecordCount() : Promise<number> {
+    const response = await fetch('http://localhost:2050/recordCount');
+    return await response.json();
+}
+
+async function getColumnNames() : Promise<string[]>{
+    const response = await fetch('http://localhost:2050/columns');
+    return await response.json();
+}
+
+async function getRecords(fromID: number, toID: number): Promise<string[][]> {
+    const response = await fetch(`http://localhost:2050/records?from=${(fromID).toString()}&to=${(toID).toString()}`);
+    return await response.json();
+}
+
+//#endregion API
+
+//#region Data Loading methods
+
+async function placeRecords(fromID: number, toID: number): Promise<number[]> {
+    const records = await getRecords(fromID, toID)
+    let appendable = '';
+    for (const record of records) {
+            appendable += `<tr id="table-row-${record[0]}">`;
+        for (const column of record) {
+            appendable += `<td align="center">${column.toString()}</td>`;     
+        }
+        appendable += '</tr>';
+    }
+    $("#WrapperTable-ContentBody").empty();
+    $("#WrapperTable-ContentBody").append(appendable);
+    return await [fromID as number, toID as number];
+}
+
+// Credit: https://gist.github.com/ca0v/73a31f57b397606c9813472f7493a940
+
+async function placeRecordsFromCursor(cursor: number[]): Promise<number[]> {
+    cursor = cursor.sort((a,b) => {return a-b});
+    return await placeRecords(cursor[0], cursor[1]);
+}
+//#endregion Data Loading methods
+
+//#region Handlers
+async function getPageContent(fromID: number, toID: number): Promise<number[]> {
+    $("#WrapperTable-HeaderRow").empty();
+    getColumnNames()
+        .then((columns: string[]) => {
+            for (const column of columns) {
+                let writable: string = `<th align="center">${column}</th>`;
+                $("#WrapperTable-HeaderRow").append(writable);
+            }
+        });
+    return await placeRecords(fromID, toID);
+}
+
+function toNumber(input: string | number, output: any = 1) : number {
+    switch (typeof input) {
+        case ('string'):
+            if (output == 1) {
+                return parseInt(input as string);
+            }
+            return parseFloat(input as string);
+        case ("number"):
+            return input as number;
+        default:
+            return 0;
+    }
+}
+
+function calculateToId(fromId: number): number {
+    const possibleRecords = Math.floor((window.innerHeight - ($("#form-content").innerHeight() as number)) / 37);
```

Comment:
Instead of casting, it is better to check for `undefined`. In this case it is safe because the div would always exist, but in our production code we would never assume this. Better to check explicitly like this:
```
const headerHeight = $("#form-content").innerHeight();
if (!headerHeight) {
    alert("Some error message");
    return;
}
const possibleRecords = Math.floor((window.innerHeight - headerHeight) / 37);
```

Casting is basically a way to get around/ignore the TypeScript type system, so only use it as a last resort.